flexmeasures/data/services/utils.py

```diff
@@ -0,0 +1,104 @@
+import functools
+from flask import current_app
+from rq.job import Job
+
+import hashlib
+import base64
+
+
+def make_hash_sha256(o):
+    """
+    SHA256 instead of Python's hash function because apparently, python native hashing function
+    yields different results on restarts.
+    Source: https://stackoverflow.com/a/42151923
+    """
+    hasher = hashlib.sha256()
+    hasher.update(repr(make_hashable(o)).encode())
+    return base64.b64encode(hasher.digest()).decode()
+
+
+def make_hashable(o):
+    """
+    Function to create hashes for dictionaries with nested objects
+    Source: https://stackoverflow.com/a/42151923
+    """
+    if isinstance(o, (tuple, list)):
+        return tuple((make_hashable(e) for e in o))
+
+    if isinstance(o, dict):
+        return tuple(sorted((k, make_hashable(v)) for k, v in o.items()))
+
+    if isinstance(o, (set, frozenset)):
+        return tuple(sorted(make_hashable(e) for e in o))
+
+    if callable(
+        getattr(o, "make_hashable", None)
+    ):  # checks if the object o has the method make_hashable
+        return o.make_hashable()
+
+    return o
+
+
+def hash_function_arguments(args, kwags):
+    """Combines the hashes of the args and kargs
+
+    The way to go to do h(x,y) = hash(hash(x) || hash(y)) because it avoid the following:
+
+    1) h(x,y) = hash(x || y), might create a collision if we delete the last n characters of x and we append them in front of y. e.g h("abc", "d") = h("ab", "cd")
+    2) we don't want to sort x and y, because we need the function h(x,y) != h(y,x)
+    3) extra hashing just avoid that we can't decompose the input arguments and track if the same args or kwarg are called several times. More of a security measure I think.
+
+    source: https://crypto.stackexchange.com/questions/55162/best-way-to-hash-two-values-into-one
+    """
+    return make_hash_sha256(
+        make_hash_sha256(args) + make_hash_sha256(kwags)
+    )  # concat two hashes
+
+
+def redis_cache(queue):
```

Comment:
This function is in my opinion not named properly. 

When I hear about a cache, I think of an [LRU cache](https://docs.python.org/3/library/functools.html#functools.cache), which works on all functions with all arguments. This is a special implementation around RQ jobs. So I guess the function could be called `job_cache()`.

And what are the difference against using an LRU cache?
- Wouldn't it technically work? Arguments are hashable, the job would be stored in the cache stack. Redundantly, gone on restarts, but technically it would, right? Just trying to understand what it really is.
- There would be no printed warning that the schedule already was computed.
- `force_new_job_creation` would not work.

If we have an extra implementation, it will pay off to document why it exists.

Reply:
Well the major benefit of using redis over LRU cache is that we can work in a distributed environment, think of multiple scheduling workers in a (Kubernetes) cluster. With the redis cache, we can avoid all workers to repeat a schedulling job when It's already been done. 

I'll add this + the reasons you mentioned in the docstring.